src/argparse.zig

```diff
@@ -0,0 +1,297 @@
+const std = @import("std");
+
+const main = @import("main");
+const NeverFailingAllocator = main.heap.NeverFailingAllocator;
+const ListUnmanaged = main.ListUnmanaged;
+
+pub fn Parser(comptime T: type, comptime callback: ?fn(self: T) anyerror!void) type {
+	return struct {
+		const Self = @This();
+		pub const Args = T;
```

Comment:
Why do we need this?

Reply:
I think it would always be more readable to have this as a separate, named, struct instead of passing it anonymously.
Note that anymous structs also tend to make names in debug information more verbose.